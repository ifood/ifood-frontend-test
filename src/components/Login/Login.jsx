import React, { useEffect } from 'react';

import { Redirect } from 'react-router-dom';

import { authEndpoint, clientId, redirectUri } from '../../utils/config';
import hash from '../../utils/getHash';

import { InnerContainer } from '../style/container/Container';
import Button from '../style/button/Button';

const Login = ({ userToken }) => {
    return (
        <InnerContainer>
            {userToken ? (
                <Redirect to="/" />
            ) : (
                <Button
                    isLink
                    href={`${authEndpoint}?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=token&show_dialog=true`}
                >
                    Logar
                </Button>
            )}
        </InnerContainer>
    );
};

export default Login;
