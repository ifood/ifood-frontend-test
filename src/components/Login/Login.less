.login {

  &__logo {
    text-align: center;
    padding: 30px 0;

    &__title {
      font-size: 21px;
      color: @light-gray;
    }
  }

  &__spotify-button{
    font-size: 18px;
  }

  &__btn-icon{
    position: relative;
    top: -3px;
  }
}
