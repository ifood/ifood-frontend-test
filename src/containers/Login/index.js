import React from 'react';
import { FaSpotify } from 'react-icons/fa';
import { useIntl } from 'react-intl';

import LayoutTemplate from '../../templates/LayoutTemplate';
import Button from '../../components/Button';
import messages from './messages';

import { getSpotifyURL } from '../../services/spotify';

const Login = () => {
  const intl = useIntl();

  const handleLogin = () => {
    const url = getSpotifyURL();
    window.location.href = url;
  };

  return (
    <LayoutTemplate>
      <Button
        value={intl.formatMessage(messages.loginButton)}
        onClick={handleLogin}
        startEnhancer={() => (
          <span>
            <FaSpotify style={{ fontSize: '1.5rem' }} />
          </span>
        )}
      />
    </LayoutTemplate>
  );
};

export default Login;
