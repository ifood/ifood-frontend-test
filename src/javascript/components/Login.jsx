import React, { Component } from "react";
import { SPOTIFY_AUTHORIZE } from "../constants/urls";
import { parseHash } from "../utils/index.js";

export default class Login extends Component {

    componentDidMount() {
        const { accessToken } = this.props;
        const hash = parseHash(window.location.hash);
        const token = hash.access_token;

        if (!accessToken && token) {
            this.props.setAccessToken(token);
        }
    }

    getLoginURL() {
        const client_id = "91a84a82e4e34192a860659488ce3ecf"
        // TODO: Make this an env variable
        const redirect_uri = "http://localhost:3000";
        const url = `${SPOTIFY_AUTHORIZE}?client_id=${client_id}&redirect_uri=${redirect_uri}&response_type=token&show_dialog=true`;

        return url;
    }

    render() {
        const { accessToken } = this.props;

        return !accessToken && (
            <div>
                <a href={this.getLoginURL()}>Login</a>
            </div>
        )
    }
}
