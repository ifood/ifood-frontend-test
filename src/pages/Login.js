import React from 'react';
import { Button, Container, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { SPOTIFY_CLIENT_ID, REDIRECT_URL } from '../constants';

const useStyles = makeStyles((theme) => ({
  button: {
    marginTop: theme.spacing(5),
  },
  h4: {
    marginTop: theme.spacing(2),
  },
  h2: {
    fontWeight: '500',
  },
  main: {
    textAlign: 'center',
    minHeight: '100vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',

  },
}));

export default function Login() {
  const classes = useStyles();
  return (
    <Container component="main" className={classes.main}>
      <Typography variant="h2" className={classes.h2}>
        SPOTIFOOD
      </Typography>
      <Typography variant="h4" className={classes.h4}>
        Top playlists para aproveitar sua comida!
      </Typography>
      <Button
        href={`https://accounts.spotify.com/authorize?client_id=${SPOTIFY_CLIENT_ID}&response_type=token&redirect_uri=${REDIRECT_URL}`}
        variant="contained"
        color="secondary"
        size="large"
        className={classes.button}
      >
        Entrar com Spotify
      </Button>
    </Container>
  );
}
