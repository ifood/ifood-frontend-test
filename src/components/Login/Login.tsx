import React from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { useAuth } from '../../hooks/useAuth';
import { useStyles } from './Login.jss';

export const Login: React.FC = () => {
  const classes = useStyles();
  const { login } = useAuth();
  return (
    <div className={classes.container}>
      <Typography className={classes.title} variant="h2">
        Spotifood
      </Typography>
      <Button color="primary" variant="contained" onClick={login} disableElevation>
        Login
      </Button>
    </div>
  );
};
