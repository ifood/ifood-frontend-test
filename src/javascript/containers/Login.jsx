import { connect } from "react-redux";
import authActions from "../actions/auth";
import Login from "../components/Login";

function mapStateToProps(state) {
    return {
        accessToken: state.auth.accessToken,
    };
}

function mapDispatchToProps(dispatch) {
    return {
        setAccessToken: token => dispatch(authActions.setAccessToken(token)),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);