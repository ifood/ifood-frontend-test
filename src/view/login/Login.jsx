import React from 'react';
import { Button, Typography } from '@material-ui/core';
import styled from 'styled-components';
import QueryString from 'query-string';
import { CenteredFlex } from '../../component/containers/Containers';
import { SCREEN, SPOTIFY_PARAMS } from '../../Constants';
import generateRandomString from '../../utils/GenerateRandomString';
import { setToken } from '../../utils/TokenController';
import spotifyLogo from '../../images/spotifood.png';

const Image = styled.img`
  zoom: 0.5;

  @media (max-width: ${SCREEN.MEDIUM}) {
    zoom: 0.3;
  }

  @media (max-width: ${SCREEN.SMALL}) {
    zoom: 0.2;
  }
`;
const BlackCenteredFlex = styled(CenteredFlex)`
  background-color: #0f0f0f;
`;
const BottomSpaced = styled.div`
  padding-bottom: 16px;
`;

function Login() {
  function handleLogin() {
    const { SPOTIFY_AUTH_RESOURCE, ...params } = SPOTIFY_PARAMS;

    const randomString = generateRandomString(16);
    const search = QueryString.stringify({ ...params, state: randomString });

    setToken(randomString);

    window.location.href = `${SPOTIFY_AUTH_RESOURCE}?${search}`;
  }

  return (
    <>
      <BlackCenteredFlex>
        <Image src={spotifyLogo} />
        <BottomSpaced>
          <Typography
            align="center"
            color="textSecondary"
            variant="h5"
            gutterBottom>
            Listen to comfy music while eating â™ª
          </Typography>
        </BottomSpaced>
        <Button variant="contained" color="primary" onClick={handleLogin}>
          Login with Spotify
        </Button>
      </BlackCenteredFlex>
    </>
  );
}

export default Login;
