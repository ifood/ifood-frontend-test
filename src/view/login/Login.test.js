/* eslint-disable no-undef, react/jsx-filename-extension, react/prop-types */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './Login';

describe('<Login />', () => {
  it('snapshot', () => {
    const wrapper = mount(<Login />);
    jestExpect(toJson(wrapper)).toMatchSnapshot();
  });
});
