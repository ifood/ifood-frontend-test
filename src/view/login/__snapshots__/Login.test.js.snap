// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c2 {
  zoom: 0.5;
}

.c1 {
  background-color: #0f0f0f;
}

.c3 {
  padding-bottom: 16px;
}

@media (max-width:700px) {
  .c2 {
    zoom: 0.3;
  }
}

@media (max-width:420px) {
  .c2 {
    zoom: 0.2;
  }
}

<Login>
  <Styled(styled.div)>
    <div
      className="c0 c1"
    >
      <styled.img
        src="spotifood.png"
      >
        <img
          className="c2"
          src="spotifood.png"
        />
      </styled.img>
      <styled.div>
        <div
          className="c3"
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            color="textSecondary"
            gutterBottom={true}
            variant="h5"
          >
            <ForwardRef(Typography)
              align="center"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              color="textSecondary"
              gutterBottom={true}
              variant="h5"
            >
              <h5
                className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
              >
                Listen to comfy music while eating â™ª
              </h5>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </styled.div>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        <ForwardRef(Button)
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit",
              "contained": "MuiButton-contained",
              "containedPrimary": "MuiButton-containedPrimary",
              "containedSecondary": "MuiButton-containedSecondary",
              "containedSizeLarge": "MuiButton-containedSizeLarge",
              "containedSizeSmall": "MuiButton-containedSizeSmall",
              "disableElevation": "MuiButton-disableElevation",
              "disabled": "Mui-disabled",
              "endIcon": "MuiButton-endIcon",
              "focusVisible": "Mui-focusVisible",
              "fullWidth": "MuiButton-fullWidth",
              "iconSizeLarge": "MuiButton-iconSizeLarge",
              "iconSizeMedium": "MuiButton-iconSizeMedium",
              "iconSizeSmall": "MuiButton-iconSizeSmall",
              "label": "MuiButton-label",
              "outlined": "MuiButton-outlined",
              "outlinedPrimary": "MuiButton-outlinedPrimary",
              "outlinedSecondary": "MuiButton-outlinedSecondary",
              "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
              "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
              "root": "MuiButton-root",
              "sizeLarge": "MuiButton-sizeLarge",
              "sizeSmall": "MuiButton-sizeSmall",
              "startIcon": "MuiButton-startIcon",
              "text": "MuiButton-text",
              "textPrimary": "MuiButton-textPrimary",
              "textSecondary": "MuiButton-textSecondary",
              "textSizeLarge": "MuiButton-textSizeLarge",
              "textSizeSmall": "MuiButton-textSizeSmall",
            }
          }
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          <WithStyles(ForwardRef(ButtonBase))
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            type="button"
          >
            <ForwardRef(ButtonBase)
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              component="button"
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="Mui-focusVisible"
              onClick={[Function]}
              type="button"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Login with Spotify
                </span>
                <WithStyles(memo)
                  center={false}
                >
                  <ForwardRef(TouchRipple)
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child",
                        "childLeaving": "MuiTouchRipple-childLeaving",
                        "childPulsate": "MuiTouchRipple-childPulsate",
                        "ripple": "MuiTouchRipple-ripple",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                        "root": "MuiTouchRipple-root",
                      }
                    }
                  >
                    <span
                      className="MuiTouchRipple-root"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component={null}
                        exit={true}
                      />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(memo)>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(Button)>
      </WithStyles(ForwardRef(Button))>
    </div>
  </Styled(styled.div)>
</Login>
`;
